lobsterpy/cohp/analyze.py

```diff
@@ -869,3 +872,240 @@ def set_summary_dicts(self):
+                "percent_kpoints_abv_limit": round(
+                    (len(dev_val) / total_kpoints) * 100, 4
+                ),
+            }  # type: ignore
+
+        else:
+            quality_dict["band_overlaps"] = {
+                "file_exists": False,
+                "limit_maxDeviation": None,
+                "has_good_quality_maxDeviation": True,
+                "max_deviation": None,
+                "percent_kpoints_abv_limit": None,
+            }  # type: ignore
+
+        if bva_comp:
+            try:
+                bond_valence = BVAnalyzer()
+
+                bva_oxi = []
+                lobs_charge = Charge(filename=path_to_charge)
+                for i in bond_valence.get_valences(structure=struct):
+                    if i >= 0:
+                        bva_oxi.append("POS")
+                    else:
+                        bva_oxi.append("NEG")
+
+                mull_oxi = []
+                for i in lobs_charge.Mulliken:
+                    if i >= 0:
+                        mull_oxi.append("POS")
+                    else:
+                        mull_oxi.append("NEG")
+
+                loew_oxi = []
+                for i in lobs_charge.Loewdin:
+                    if i >= 0:
+                        loew_oxi.append("POS")
+                    else:
+                        loew_oxi.append("NEG")
+
+                quality_dict["Charges"] = {}  # type: ignore
+                if mull_oxi == bva_oxi:
+                    quality_dict["Charges"]["BVA_Mulliken_agree"] = True  # type: ignore
+                else:
+                    quality_dict["Charges"]["BVA_Mulliken_agree"] = False  # type: ignore
+
+                if mull_oxi == bva_oxi:
+                    quality_dict["Charges"]["BVA_Loewdin_agree"] = True  # type: ignore
+                else:
+                    quality_dict["Charges"]["BVA_Loewdin_agree"] = False  # type: ignore
+            except ValueError:
+                quality_dict["Charges"] = {}  # type: ignore
+                warnings.warn(
+                    "Oxidation states from BVA analyzer cannot be determined. "
+                    "Thus BVA charge comparison will be skipped"
+                )
+        if dos_comparison:
+            if "LSO" not in str(path_to_doscar).split("."):
+                warnings.warn(
+                    "Consider using DOSCAR.LSO.lobster, as non LSO DOS from LOBSTER can have "
+                    "negative DOS values"
+                )
+            doscar_lobster = Doscar(
+                doscar=path_to_doscar,
+                structure_file=path_to_poscar,
+            )
+
+            dos_lobster = doscar_lobster.completedos
+
+            vasprun = Vasprun(path_to_vasprun)
+            dos_vasp = vasprun.complete_dos
+
+            quality_dict["DOS_comparisons"] = {}  # type: ignore
+
+            for orb in dos_lobster.get_spd_dos():
+                if e_range[0] >= min(dos_vasp.energies) and e_range[0] >= min(

```

Comment:
Your test cases cover all these ifs?

Reply:
Updated tests and now if covers all this 😃